client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/RssShuffleManagerBase.java

```diff
@@ -715,12 +719,19 @@ public boolean reassignOnStageResubmit(
   public MutableShuffleHandleInfo reassignOnBlockSendFailure(
       int shuffleId, Map<Integer, List<ReceivingFailureServer>> partitionToFailureServers) {
     long startTime = System.currentTimeMillis();
-    MutableShuffleHandleInfo handleInfo =
-        (MutableShuffleHandleInfo) shuffleHandleInfoManager.get(shuffleId);
-    synchronized (handleInfo) {
+    ShuffleHandleInfo handleInfo = shuffleHandleInfoManager.get(shuffleId);
+    MutableShuffleHandleInfo internalHandle = null;
+    if (handleInfo instanceof MutableShuffleHandleInfo) {
+      internalHandle = (MutableShuffleHandleInfo) handleInfo;
+    } else if (handleInfo instanceof StageAttemptShuffleHandleInfo) {
+      internalHandle =
+          (MutableShuffleHandleInfo) ((StageAttemptShuffleHandleInfo) handleInfo).getCurrent();
+    }
+    assert internalHandle != null;
```

Comment:
We use throw an exception here instead of using assertions. See https://github.com/apache/incubator-uniffle/issues/1668.

Reply:
Sounds good.